From c130ec0e1b3a41a7387bef1489e49e98df9fd7f5 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Fri, 19 May 2023 17:30:21 +0200
Subject: [PATCH 0674/1695] plat-stm32mp2: conf: enable SAES for stm32mp2
 platforms

Default enables stm32_saes and crypto framework for stm32mp2 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0cb3a78a8eb1a26ed8e876952131a197be06039c plat-stm32mp2: conf: enable SAES for stm32mp2 platforms]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ia2dfcdb6373e34b86832118704afb902bd71dfe2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361856
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 2d8a8468d..8944aa5d2 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -141,6 +141,7 @@ CFG_STM32_RISAB ?= y
 CFG_STM32_RISAF ?= y
 CFG_STM32_RNG ?= y
 CFG_STM32_RTC ?= y
+CFG_STM32_SAES ?= y
 CFG_STM32_SERC ?= y
 CFG_STM32_STGEN ?= y
 CFG_STM32_TAMP ?= y
@@ -172,6 +173,11 @@ ifeq ($(CFG_STM32_HSE_MONITORING),y)
 $(call force,CFG_STM32_LPTIMER,y)
 endif
 
+# If any crypto driver is enabled, enable the crypto-framework layer
+ifeq ($(call cfg-one-enabled, CFG_STM32_SAES),y)
+$(call force,CFG_STM32_CRYPTO_DRIVER,y)
+endif
+
 CFG_STM32MP_REMOTEPROC ?= n
 CFG_DRIVERS_REMOTEPROC ?= $(CFG_STM32MP_REMOTEPROC)
 CFG_REMOTEPROC_PTA ?= $(CFG_STM32MP_REMOTEPROC)
-- 
2.47.2

